src/Microsoft.ComponentDetection.Contracts/TypedComponent/TypedComponent.cs

```diff
@@ -30,6 +30,10 @@ internal TypedComponent()
     [JsonIgnore]
     internal string DebuggerDisplay => $"{this.Id}";
 
+    [field: JsonIgnore]
+    [JsonIgnore]
+    protected string CacheId { get; set; }
```

Comment:
Another suggestion that I have is instead of this being a property we can have it as a protected attribute in the class `id`. and keep the Id property as the method that returns the attribute. We can keep the `cached` concept abstract. Another change that we can do but require more changes is to defined the properties that conform the id as required and we can then create the id when the object is created.

Reply:
I think this is what mpysson was talking about.